snap/squashfs/squashfs.go

```diff
@@ -273,6 +275,131 @@ func (s *Snap) ReadFile(filePath string) (content []byte, err error) {
 	return content, nil
 }
 
+var sdVer int
+
+func init() {
+	// this also avoids calling systemctl all the time which perturbs
+	// some unrelated tests
+	v, err := systemd.Version()
+	if err != nil {
+		v = 0
+	}
+	sdVer = v
+}
+
+var (
+	sysGeteuid = sys.Geteuid
+)
+
+func sandboxParams(sdVer int) (params []string) {
+	if sysGeteuid() != 0 {
+		return nil
+	}
+	// effetive user is 0, use systemd-run to sandbox unsquashfs
+	params = []string{
+		"--property=KeyringMode=private",
+		"--property=PrivateTmp=true",
+		// xxx CAP_DAC_OVERRIDE is mainly possibly needed for
+		// prepare-image run under sudo
+		"--property=CapabilityBoundingSet=CAP_DAC_OVERRIDE",
+		"--property=AmbientCapabilities=CAP_DAC_OVERRIDE",
+		"--property=NoNewPrivileges=true",
+		"--property=SecureBits=",
+		"--property=ProtectSystem=strict",
+		"--property=ProtectHome=true",
+		"--property=ProtectKernelTunables=true",
+		"--property=ProtectControlGroups=true",
+		"--property=ProtectKernelModules=true",
+		"--property=PrivateDevices=true",
+		"--property=MemoryDenyWriteExecute=true",
+		"--property=RestrictSUIDSGID=true",
+		"--property=SystemCallFilter=@default @basic-io @signal @file-system @chown @process mprotect",
+		"--property=SystemCallErrorNumber=EPERM",
+		"--property=MemoryMax=16M",
```

Comment:
Is MemoryMax=16M too restrictive? Could this cause unsquashfs to fail on OOM?

Reply:
I don't think it reads entire files into memory, so I would expect it be ok with this. I had it work properly going down to 2M for example.

Reply:
Ok no worries.